src/process_utility.c

```diff
@@ -3210,6 +3210,56 @@ process_altertable_chunk(Hypertable *ht, Oid chunk_relid, void *arg)
 	AlterTableInternal(chunk_relid, list_make1(cmd), false);
 }
 
+static void
+process_altertable_chunk_replica_identity(Hypertable *ht, Oid chunk_relid, void *arg)
+{
+	AlterTableCmd *cmd = castNode(AlterTableCmd, copyObject(arg));
+	ReplicaIdentityStmt *stmt = castNode(ReplicaIdentityStmt, cmd->def);
+
+	if (stmt->identity_type == REPLICA_IDENTITY_INDEX)
+	{
+		Chunk *chunk = ts_chunk_get_by_relid(chunk_relid, true);
+		Oid hyper_schema_oid = get_rel_namespace(ht->main_table_relid);
+		Oid hyper_index_oid = get_relname_relid(stmt->name, hyper_schema_oid);
+		ChunkIndexMapping cim;
+
+		Assert(OidIsValid(hyper_index_oid));
+
+		if (!ts_chunk_index_get_by_hypertable_indexrelid(chunk, hyper_index_oid, &cim))
+			elog(ERROR,
+				 "chunk \"%s\" has no index corresponding to hypertable index \"%s\"",
+				 get_rel_name(chunk_relid),
+				 stmt->name);
+
+		stmt->name = get_rel_name(cim.indexoid);
+	}
+
+	AlterTableInternal(chunk_relid, list_make1(cmd), false);
+}
+
+static void
+process_altertable_replica_identity(Hypertable *ht, AlterTableCmd *cmd)
+{
+	ReplicaIdentityStmt *stmt = castNode(ReplicaIdentityStmt, cmd->def);
+
+	if (stmt->identity_type == REPLICA_IDENTITY_INDEX)
+	{
+		Oid hyper_schema_oid = get_rel_namespace(ht->main_table_relid);
+		Oid hyper_index_oid;
+
+		hyper_index_oid = get_relname_relid(stmt->name, hyper_schema_oid);
+
+		if (!OidIsValid(hyper_index_oid))
+			ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_OBJECT),
+					 errmsg("index \"%s\" for table \"%s\" does not exist",
+							stmt->name,
+							get_rel_name(ht->main_table_relid))));
```

Comment:
We are printing full names elsewhere.
```suggestion
					 errmsg("index \"%s\" for table \"%s.%s\" does not exist",
							stmt->name,
							get_namespace_name(hyper_schema_oid),
							get_rel_name(ht->main_table_relid))));
```